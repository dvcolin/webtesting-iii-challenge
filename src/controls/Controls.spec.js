import React from 'react'
import { render, fireEvent } from 'react-testing-library'
import renderer from 'react-test-renderer'

import Controls from './Controls'

describe('<Controls />', () => {
    //Snapshot test
    it('should match snapshot', () => {
        const tree = renderer.create(<Controls />);

        expect(tree.toJSON()).toMatchSnapshot();
    })

    it('should toggle open/close when open button is clicked', () => {
        const { getByTestId } = render(<Controls />);
        const openButton = getByTestId('open-button');
        
        fireEvent.click(openButton);
    })
    
    it('should toggle lock/unlock when lock button is clicked', () => {
        const { getByTestId } = render(<Controls />);
        const lockButton = getByTestId('lock-button');
        
        fireEvent.click(lockButton);
    })
});