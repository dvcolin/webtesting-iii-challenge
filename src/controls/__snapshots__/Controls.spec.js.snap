// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Controls /> should match snapshot 1`] = `
<div
  className="controls panel"
>
  <button
    className="toggle-btn"
    data-testid="lock-button"
    disabled={true}
  >
    Lock Gate
  </button>
  <button
    className="toggle-btn"
    data-testid="open-button"
  >
    Close Gate
  </button>
</div>
`;
